include/util/geojson_debug_logger.hpp

```diff
@@ -63,12 +63,8 @@ class GeojsonLogger
         // make sure to syncronize logging output, our writing should be sequential
         std::lock_guard<std::mutex> guard(lock);
 
-        // if there is no logfile, we cannot write
-        if (!ofs.is_open() || (nullptr == policy))
-        {
-            // this can only happend between two guards when concurrent writing occurs
-            return false;
-        }
+        // if there is no logfile, we cannot write (possible reason: the guard might be out of scope (e.g. if it is anonymous))
+        BOOST_ASSERT_MSG(!ofs.is_open() || (nullptr == policy), "Trying to use the geojson printer without an open logger.")
```

Comment:
Are we actually planning to use the geojson printer for anything else but debugging? If it is purely a debugging tool we might not want to throw? Or should we throw?